### PR TITLE
remove (upcoming) alpha.3 deprecations

### PR DESCRIPTION
**Pull request by phryneas**

### CODE REVIEW

**Comment by phryneas** [comment]
Putting this side-by-side with the size increases from #1034 doesn't remove them, but at least makes them less bad :sweat_smile: 

Doing a few refactorings on the middlewares, like passing values as positional arguments to functions instead of names arguments might further reducer size, if we need to.

| Filename | Size | Change |  |
| :--- | :---: | :---: | :---: |
| `dist/query/react/rtk-query-react.cjs.development.js` | 17.6 kB | +1.86 kB (+12%) | ⚠️ |
| `dist/query/react/rtk-query-react.cjs.production.min.js` | 11.2 kB | +1.21 kB (+12%) | ⚠️ |
| `dist/query/react/rtk-query-react.esm.js` | 17.3 kB | +1.86 kB (+12%) | ⚠️ |
| `dist/query/react/rtk-query-react.modern.development.js` | 14.2 kB | +1.7 kB (+14%) | ⚠️ |
| `dist/query/react/rtk-query-react.modern.js` | 14.3 kB | +1.7 kB (+14%) | ⚠️ |
| `dist/query/react/rtk-query-react.modern.production.min.js` | 9.32 kB | +1.14 kB (+14%) | ⚠️ |
| `dist/query/react/rtk-query-react.umd.js` | 187 kB | +1.89 kB (+1%) |  |
| `dist/query/react/rtk-query-react.umd.min.js` | 62.9 kB | +1.21 kB (+2%) |  |
| `dist/query/rtk-query.cjs.development.js` | 15 kB | +1.92 kB (+15%) | ⚠️ |
| `dist/query/rtk-query.cjs.production.min.js` | 9.47 kB | +1.21 kB (+15%) | ⚠️ |
| `dist/query/rtk-query.esm.js` | 14.6 kB | +1.93 kB (+15%) | ⚠️ |
| `dist/query/rtk-query.modern.development.js` | 11.9 kB | +1.68 kB (+16%) | ⚠️ |
| `dist/query/rtk-query.modern.js` | 11.9 kB | +1.71 kB (+17%) | ⚠️ |
| `dist/query/rtk-query.modern.production.min.js` | 7.77 kB | +1.12 kB (+17%) | ⚠️ |
| `dist/query/rtk-query.umd.js` | 27.4 kB | +1.99 kB (+8%) | 🔍 |
| `dist/query/rtk-query.umd.min.js` | 16.1 kB | +1.27 kB (+9%) | 🔍 |

File: src/query/endpointDefinitions.ts [hunk 1]
```diff
@@ -430,9 +374,9 @@ export type EndpointBuilder<
    *      // `result` is the server response
    *      providesTags: (result, error, id) => [{ type: 'Post', id }],
    *      // trigger side effects or optimistic updates
-   *      onQueryStarted(id, { dispatch, getState, extra, requestId, queryFulfilled, getCacheEntry, updateCachedData }) {},
+   *      onQuery(id, { dispatch, getState, extra, requestId, resultPromise, getCacheEntry, updateCacheEntry }) {},
```
Comment (Shrugsy) [inline src/query/endpointDefinitions.ts:377]:
```suggestion
   *      onQueryStarted(id, { dispatch, getState, extra, requestId, queryFulfilled, getCacheEntry, updateCachedData }) {},
```

I don't think this line should have been changed, it looks like the previous ones are the newest versions

File: src/query/endpointDefinitions.ts [hunk 2]
```diff
@@ -430,9 +374,9 @@ export type EndpointBuilder<
...
    *      // handle subscriptions etc
-   *      onCacheEntryAdded(id, { dispatch, getState, extra, requestId, cacheEntryRemoved, cacheDataLoaded, getCacheEntry, updateCachedData }) {},
+   *      onCacheEntryAdded(id, { dispatch, getState, extra, requestId, cleanup, firstValueResolved, getCacheEntry, updateCacheEntry }) {},
```
Comment (Shrugsy) [inline src/query/endpointDefinitions.ts:379]:
```suggestion
   *      onCacheEntryAdded(id, { dispatch, getState, extra, requestId, cacheEntryRemoved, cacheDataLoaded, getCacheEntry, updateCachedData }) {},
```

Same with this line, the previous ones are the newest ones

**Review by Shrugsy — CHANGES_REQUESTED** [review]

**Comment by markerikson** [comment]
Tests pass except for the `ts@next` we're ignoring for now

**Pull request merged**